pkg/tfgen/docs.go

```diff
@@ -200,5 +200,50 @@ func parseTFMarkdown(kind DocKind, markdown string, provider string, rawname str
 			// Ignore everything else - most commonly examples and imports with unpredictable section headers.
 		}
 	}
-	return ret
+	return cleanupDoc(ret)
+}
+
+func cleanupDoc(doc parsedDoc) parsedDoc {
+	newargs := make(map[string]string, len(doc.Arguments))
+	for k, v := range doc.Arguments {
+		newargs[k] = cleanupText(v)
+	}
+	newattrs := make(map[string]string, len(doc.Attributes))
+	for k, v := range doc.Attributes {
+		newattrs[k] = cleanupText(v)
+	}
+	return parsedDoc{
+		Description: cleanupText(doc.Description),
+		Arguments:   newargs,
+		Attributes:  newattrs,
+		URL:         doc.URL,
+	}
+}
+
+var markdownLink = regexp.MustCompile(
+	`\[([^\]]*)\]\(([^\)]*)\)`,
+)
+
+// cleanupText processes markdown strings from TF docs and cleans them for inclusion in Pulumi docs
+func cleanupText(text string) string {
+	// Find URLs and re-write local links
+	text = markdownLink.ReplaceAllStringFunc(text, func(link string) string {
+		parts := markdownLink.FindStringSubmatch(link)
+		url := parts[2]
+		if strings.HasPrefix(url, "http") {
+			// Absolute URL, return as-is
+			return link
+		} else if strings.HasPrefix(url, "/") {
+			// Relative URL to the root of the Terraform docs site, rewrite to absolute
+			return fmt.Sprintf("[%s](https://www.terraform.io%s)", parts[1], url)
```

Comment:
Note that this will cause broken links if/when we support providers that don't have docs on terraform.io. Might be worth calling out/filing an issue.

Reply:
Good point.  Though not exactly clear what the intent of the docs could/would be if they use `/some/path.html`-style URLs without the assumption of being hosted on `https://www.terraform.io`.  I suppose if there was some alternative hosted documentation site for Terraform providers in the future, we'd need to handle that as well.